src/components/productList/ProductList.component.tsx

```diff
@@ -1,24 +1,96 @@
+import { useEffect, useState } from 'react';
+
+import FiltersComponent from '@/components/filters/Filters.component.tsx';
 import type { Product } from '@/interfaces/Products.ts';
 import mockProducts from '@/mocks/mockData.json';
 
+import PaginationComponent from '../pagination/Pagination.component.tsx';
+
 import ProductCard from './ProductCard.component.tsx';
 
 import styles from './ProductList.module.css';
 
-interface ProductsListProps {
-    addToCart: () => void;
-    theme: 'light' | 'dark';
-}
-const ProductsList: React.FC<ProductsListProps> = ({ addToCart, theme }) => (
-    <>
-        <div className={styles.container}>
-            <div className={styles.cardsPlacement}>
-                {mockProducts.map((product: Product) => (
-                    <ProductCard key={product.id} product={product} addToCart={addToCart} theme={theme} />
-                ))}
+const ProductsList: React.FC = () => {
+    const [currentPage, setCurrentPage] = useState(1);
+    const [productsPerPage] = useState(8);
+    const [filteredProducts, setFilteredProducts] = useState<Product[]>([]);
+    const [selectedCategories, setSelectedCategories] = useState<string[]>([]);
+    const [sortOption, setSortOption] = useState<string>('priceSort');
+    const [searchQuery, setSearchQuery] = useState<string>('');
+
+    useEffect(() => {
+        const filterAndSortProducts = () => {
```

Comment:
split this into 2 functions. Generally, your function has to do only one think. You can learn more about Single Responsibility principle.